### PR TITLE
PULSEBS-67 story implementation

### PR DESCRIPTION
**Pull request by Francexi**

### CODE REVIEW

File: server/API/REST.php [hunk 1]
```diff
@@ -552,12 +559,13 @@ function book_lecture($vars) {
 			}
 
 			// Success
+			$in_wait_list = check_user_in_waiting_list($lectureId);
 			//I send a confirmazion email
 			$mail_result = mail($user_data["email"], "Confirmation of " . $lecture["name"] . " lecture booking", "You did a booking for the lecture of " . $lecture["name"] . ". The booking has been successfull");
```
Comment (LeaX-XIV) [inline server/API/REST.php]: Mention in the mail if student is in waiting list
```suggestion
			//I send a confirmation email
			$mail_subject = "Confirmation of " . $lecture["name"] . " lecture booking";
			$mail_body = "You succesfully booked for the lecture of " . $lecture["name"] .". ($in_wait_list ? " The room is currently at full capacity, you have been placed in a waiting list." : "");
			$mail_result = mail($user_data["email"], $mail_subject, $mail_body);
```

File: server/functions.php [hunk 2]
```diff
@@ -33,4 +33,67 @@ function check_login() {
...
+			throw new PDOException($stmt->errorInfo()[2]);
+		}
+
+		$seats = $stmt->fetchColumn();
+
+		if($seats === false){
+			throw new ErrorException("Cannot retrieve number of seats");
+		}
+
+		return $seats;
```
Comment (LeaX-XIV) [inline server/functions.php]:
```suggestion
		return intval($seats);
```

File: server/functions.php [hunk 3]
```diff
@@ -33,4 +33,67 @@ function check_login() {
...
+		$stmt = $pdo->prepare("SELECT user_id FROM bookings WHERE lecture_id = :lectureId AND cancellation_ts IS NULL ORDER_BY booking_ts ASC LIMIT -1 OFFSET :seats");
+		
+		$stmt->bindValue(":lectureId", $lecture_id, PDO::PARAM_INT);
+		$stmt->bindValue(":seats", $seats, PDO::PARAM_INT);
+
+		if (!$stmt->execute()) {
+			throw new PDOException($stmt->errorInfo()[2]);
+		}
+
+		return $stmt->fetchAll(PDO::FETCH_COLUMN, 0);
```
Comment (LeaX-XIV) [inline server/functions.php:96]: Is this [string] or [int]?

File: server/API/REST.php (hunk 1 above)
Comment (Francexi) [reply server/API/REST.php]: good idea

**Review by LeaX-XIV — APPROVED** [review]

**Pull request merged**